feat_: `LogOnPanic` linter

igor-sirotin opened the pull request:
Closes https://github.com/status-im/status-go/issues/5866

FYI: I know it's not an urgent task. 
I mostly used my free time to play with it, because I was interested in implementing such a thing.

> [!IMPORTANT]
> Review by commits

- 204c40a46 chore_: make vendor
- a5bca14ff fix_: add missing defer LogOnPanic
- ed22a12db feat_: LogOnPanic linter

# Description

https://github.com/status-im/status-go/pull/5849 introduced a requirement for `defer LogOnPanic()` to be as the first line in all goroutines in status-go code. This PR introduces a custom linter that checks if the requirement is satisifted.

3 cases are supported:
- `go func() { ... }`
- `go <free-func>`
- `go <method>`

~~So far it seems that only [1 case is not supported](https://github.com/status-im/status-go/pull/5849#discussion_r1768453322): when a goroutine is passed as an argument. But after implementing this PR, it seems to me that it should be very easy to track as well, we have access to all `go` statements. 
UPD: I even think that it might be working right in this PR 🤔~~

UPD2: This case is also kinda covered by the linter, just not directly.
It will detect such go statement:
```
func runAsync(fn func()) {
	go fn() // want "missing defer call to LogOnPanic"
}
```
... it just won't be able to check if the body of `fn` has the `LogOnPanic`.
But if you wrap `fn` into an anonymous function, then we're back to a usual case of anonymous funcs and the linter is satisfied.
```
func runAsyncOk(fn func()) {
	go func() {
		defer common.LogOnPanic()
		fn()
	}()
}
```

So far there's only one such case in our code and it already uses an anonymous func wrapper 👌 
https://github.com/status-im/status-go/blob/38308d48f2fb26058b599c9c46c0eb212203df0c/api/utils.go#L13-L23

# Implementation

The linter uses `go/analysis` and implements the `Analyzer` type. 
It also runs `gopls` under the hood to find definitions of funcs/methods. 

Later this could be integrated into `golangci-lint`, if needed.

# Example

Here's what you get without the fixes in a3382f1a8:

```go
/Users/igorsirotin/Repositories/Status/status-go/waku/v1/peer.go:87:7: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:199:11: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:1222:17: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:1721:22: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/protocol/encryption/publisher/publisher.go:54:7: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/healthmanager/blockchain_health_manager.go:74:5: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/rpc/client.go:178:7: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/services/wallet/balance/ttl_cache.go:59:13: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/services/wallet/api.go:781:5: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/services/wallet/api.go:843:5: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/services/wallet/api.go:929:7: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/protocol/messenger.go:584:22: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/protocol/messenger.go:919:28: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/protocol/messenger_store_node_request_manager.go:563:7: missing defer call to LogOnPanic
/Users/igorsirotin/Repositories/Status/status-go/services/wallet/common/mock/feed_subscription.go:22:5: missing defer call to LogOnPanic
```

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (78)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 0bb7cc86 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/1/) | 2024-10-19 19:10:25 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/1//consoleText) |
| :heavy_check_mark: | 0bb7cc86 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/1/) | 2024-10-19 19:13:18 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241019-190839-0bb7cc-pr5969.zip) |
| :heavy_check_mark: | 0bb7cc86 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/1/) | 2024-10-19 19:13:24 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241019-190840-0bb7cc-pr5969.zip) |
| :heavy_check_mark: | 0bb7cc86 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/1/) | 2024-10-19 19:13:49 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241019-190838-0bb7cc-pr5969.aar) |
| :heavy_check_mark: | 0bb7cc86 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/1/) | 2024-10-19 19:14:21 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/1//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 0ac1275c | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/2/) | 2024-10-19 19:18:45 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/2//consoleText) |
| :heavy_check_mark: | 0ac1275c | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/2/) | 2024-10-19 19:18:53 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241019-191709-0ac127-pr5969.aar) |
| :heavy_check_mark: | 0ac1275c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/2/) | 2024-10-19 19:19:36 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241019-191712-0ac127-pr5969.zip) |
| :heavy_check_mark: | 0ac1275c | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/2/) | 2024-10-19 19:20:07 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241019-191711-0ac127-pr5969.zip) |
| :heavy_check_mark: | 0ac1275c | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/2/) | 2024-10-19 19:22:40 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4802b92e | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/3/) | 2024-10-19 19:34:14 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241019-193230-4802b9-pr5969.aar) |
| :heavy_multiplication_x: | 4802b92e | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/3/) | 2024-10-19 19:35:44 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/3//consoleText) |
| :heavy_check_mark: | 4802b92e | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/3/) | 2024-10-19 19:36:57 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241019-193235-4802b9-pr5969.zip) |
| :heavy_check_mark: | 4802b92e | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/3/) | 2024-10-19 19:37:47 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/3//consoleText) |
| :heavy_check_mark: | 4802b92e | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/3/) | 2024-10-19 19:39:49 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241019-193231-4802b9-pr5969.zip) |
| | | | | | | |
| :heavy_multiplication_x: | fd29ab5c | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/4/) | 2024-10-22 15:35:59 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/4//consoleText) |
| :heavy_check_mark: | fd29ab5c | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/4/) | 2024-10-22 15:37:06 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-153204-fd29ab-pr5969.zip) |
| :heavy_check_mark: | fd29ab5c | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/4/) | 2024-10-22 15:37:23 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-153207-fd29ab-pr5969.zip) |
| :heavy_check_mark: | fd29ab5c | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/4/) | 2024-10-22 15:38:40 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-153203-fd29ab-pr5969.aar) |
| :heavy_multiplication_x: | fd29ab5c | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/4/) | 2024-10-22 15:38:59 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/4//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 5a2c129f | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/5/) | 2024-10-22 15:38:54 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | f42f8c97 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/5/) | 2024-10-22 15:40:44 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-153725-f42f8c-pr5969.zip) |
| :heavy_check_mark: | f42f8c97 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/5/) | 2024-10-22 15:40:51 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-153738-f42f8c-pr5969.zip) |
| :heavy_check_mark: | f42f8c97 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/5/) | 2024-10-22 15:42:02 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-153905-f42f8c-pr5969.aar) |
| | | | | | | |
| :heavy_check_mark: | 67bfb7ee | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/6/) | 2024-10-22 15:44:00 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-154056-67bfb7-pr5969.zip) |
| :heavy_multiplication_x: | 67bfb7ee | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/7/) | 2024-10-22 15:44:05 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/7//consoleText) |
| :heavy_check_mark: | 67bfb7ee | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/6/) | 2024-10-22 15:44:30 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-154109-67bfb7-pr5969.zip) |
| :heavy_check_mark: | 67bfb7ee | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/6/) | 2024-10-22 15:48:04 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/6//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 6a7d004c | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/8/) | 2024-10-22 15:46:27 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/8//consoleText) |
| :heavy_check_mark: | 6a7d004c | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/7/) | 2024-10-22 15:46:35 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-154252-6a7d00-pr5969.aar) |
| :heavy_check_mark: | 6a7d004c | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/7/) | 2024-10-22 15:47:00 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-154413-6a7d00-pr5969.zip) |
| :heavy_check_mark: | 6a7d004c | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/7/) | 2024-10-22 15:48:00 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-154444-6a7d00-pr5969.zip) |
| :heavy_multiplication_x: | 6a7d004c | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/7/) | 2024-10-22 15:54:26 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 52ba0682 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/8/) | 2024-10-22 17:16:23 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-171422-52ba06-pr5969.aar) |
| :heavy_multiplication_x: | 52ba0682 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/9/) | 2024-10-22 17:17:36 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/9//consoleText) |
| :heavy_check_mark: | 52ba0682 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/8/) | 2024-10-22 17:17:44 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-171423-52ba06-pr5969.zip) |
| :heavy_check_mark: | 52ba0682 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/8/) | 2024-10-22 17:19:32 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-171427-52ba06-pr5969.zip) |
| :heavy_multiplication_x: | 52ba0682 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/8/) | 2024-10-22 17:20:36 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/8//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 91939931 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/10/) | 2024-10-22 18:57:51 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/10//consoleText) |
| :heavy_check_mark: | 91939931 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/9/) | 2024-10-22 18:57:53 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-185520-919399-pr5969.zip) |
| :heavy_check_mark: | 91939931 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/9/) | 2024-10-22 18:57:57 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-185517-919399-pr5969.zip) |
| :heavy_check_mark: | 91939931 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/9/) | 2024-10-22 18:58:19 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-185516-919399-pr5969.aar) |
| :heavy_multiplication_x: | 91939931 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/9/) | 2024-10-22 19:01:09 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | bd6c6d96 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/10/) | 2024-10-22 19:12:34 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-191024-bd6c6d-pr5969.aar) |
| :heavy_check_mark: | bd6c6d96 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/10/) | 2024-10-22 19:13:05 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-191027-bd6c6d-pr5969.zip) |
| :heavy_check_mark: | bd6c6d96 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/10/) | 2024-10-22 19:13:26 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-191025-bd6c6d-pr5969.zip) |
| :heavy_multiplication_x: | bd6c6d96 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/11/) | 2024-10-22 19:14:25 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/11//consoleText) |
| :heavy_multiplication_x: | bd6c6d96 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/10/) | 2024-10-22 19:16:46 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/10//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | a3382f1a | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/12/) | 2024-10-22 21:02:16 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/12//consoleText) |
| :heavy_check_mark: | a3382f1a | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/11/) | 2024-10-22 21:02:26 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-210038-a3382f-pr5969.aar) |
| :heavy_check_mark: | a3382f1a | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/11/) | 2024-10-22 21:03:38 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-210039-a3382f-pr5969.zip) |
| :heavy_check_mark: | a3382f1a | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/11/) | 2024-10-22 21:05:10 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-210042-a3382f-pr5969.zip) |
| :heavy_multiplication_x: | a3382f1a | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/11/) | 2024-10-22 21:07:06 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 204c40a4 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/12/) | 2024-10-22 21:07:21 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-210502-204c40-pr5969.aar) |
| :heavy_check_mark: | 204c40a4 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/12/) | 2024-10-22 21:07:47 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-210524-204c40-pr5969.zip) |
| :heavy_check_mark: | 204c40a4 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/12/) | 2024-10-22 21:08:03 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-210503-204c40-pr5969.zip) |
| :heavy_multiplication_x: | 204c40a4 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/13/) | 2024-10-22 21:08:59 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/13//consoleText) |
| :heavy_check_mark: | 204c40a4 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/12/) | 2024-10-22 21:12:47 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/12//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b0c689ad | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/13/) | 2024-10-22 21:16:45 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-211458-b0c689-pr5969.aar) |
| :heavy_multiplication_x: | b0c689ad | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/14/) | 2024-10-22 21:17:04 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/14//consoleText) |
| :heavy_check_mark: | b0c689ad | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/13/) | 2024-10-22 21:17:46 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-211502-b0c689-pr5969.zip) |
| :heavy_check_mark: | b0c689ad | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/13/) | 2024-10-22 21:18:52 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-211459-b0c689-pr5969.zip) |
| :heavy_multiplication_x: | b0c689ad | [#13](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/13/) | 2024-10-22 21:21:03 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/13//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2362d1df | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/14/) | 2024-10-22 21:39:22 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241022-213736-2362d1-pr5969.aar) |
| :heavy_check_mark: | 2362d1df | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/14/) | 2024-10-22 21:40:15 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241022-213737-2362d1-pr5969.zip) |
| :heavy_check_mark: | 2362d1df | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/14/) | 2024-10-22 21:40:24 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241022-213740-2362d1-pr5969.zip) |
| :heavy_multiplication_x: | 2362d1df | [#14](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/14/) | 2024-10-22 21:43:41 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/14//consoleText) |
| :heavy_multiplication_x: | 2362d1df | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/15/) | 2024-10-22 22:10:06 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/15//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 304c0498 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/15/) | 2024-10-23 09:04:32 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241023-090238-304c04-pr5969.aar) |
| :heavy_check_mark: | 304c0498 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/15/) | 2024-10-23 09:05:24 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241023-090243-304c04-pr5969.zip) |
| :heavy_check_mark: | 304c0498 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/15/) | 2024-10-23 09:05:34 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241023-090239-304c04-pr5969.zip) |
| :heavy_multiplication_x: | 304c0498 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/15/) | 2024-10-23 09:08:40 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/15//consoleText) |
| :heavy_check_mark: | 304c0498 | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/16/) | 2024-10-23 09:35:04 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/16//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 9de1a4b8 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/16/) | 2024-10-23 10:02:53 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241023-100104-9de1a4-pr5969.aar) |
| :heavy_check_mark: | 9de1a4b8 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/16/) | 2024-10-23 10:03:50 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241023-100109-9de1a4-pr5969.zip) |
| :heavy_check_mark: | 9de1a4b8 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/16/) | 2024-10-23 10:04:14 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241023-100106-9de1a4-pr5969.zip) |
| :heavy_multiplication_x: | 9de1a4b8 | [#16](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/16/) | 2024-10-23 10:07:10 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/16//consoleText) |
| :heavy_check_mark: | 9de1a4b8 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/17/) | 2024-10-23 10:33:28 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/17//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 7735ef97 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/17/) | 2024-10-23 11:39:16 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241023-113643-7735ef-pr5969.zip) |
| :heavy_check_mark: | 7735ef97 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/17/) | 2024-10-23 11:42:11 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241023-113646-7735ef-pr5969.zip) |
| :heavy_check_mark: | 7735ef97 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/17/) | 2024-10-23 11:42:34 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/17//consoleText) |
| :heavy_check_mark: | 7735ef97 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/17/) | 2024-10-23 11:43:13 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241023-113642-7735ef-pr5969.aar) |
| :heavy_check_mark: | 7735ef97 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/18/) | 2024-10-23 12:09:08 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/18//consoleText) |
| | | | | | | |
| :heavy_check_mark: | cc809f92 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5969/18/) | 2024-10-23 19:41:18 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241023-193855-cc809f-pr5969.aar) |
| :heavy_check_mark: | cc809f92 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5969/18/) | 2024-10-23 19:41:37 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241023-193859-cc809f-pr5969.zip) |
| :heavy_check_mark: | cc809f92 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5969/18/) | 2024-10-23 19:42:03 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241023-193857-cc809f-pr5969.zip) |
| :heavy_check_mark: | cc809f92 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/18/) | 2024-10-23 19:44:55 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5969/18//consoleText) |
| :heavy_check_mark: | cc809f92 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/19/) | 2024-10-23 20:11:32 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5969/19//consoleText) |

igor-sirotin committed:
feat_: LogOnPanic linter

igor-sirotin committed:
chore_: make vendor


make vendor

igor-sirotin committed:
fix_: add missing defer LogOnPanic

igor-sirotin commented:
Results after fixing existing issues, but before filtering out `./cmd` directory.

```
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/spiff-workflow/main.go:144:6: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/spiff-workflow/main.go:145:18: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/spiff-workflow/main.go:443:5: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/spiff-workflow/main.go:470:5: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/status-cli/message.go:195:5: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/status-cli/serve.go:111:6: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/status-cli/simulate.go:19:5: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/status-cli/simulate.go:68:11: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/status-cli/simulate.go:70:13: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/statusd/main.go:181:10: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/statusd/main.go:284:6: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/statusd/main.go:300:7: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/statusd/main.go:310:7: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/statusd/main.go:311:19: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/statusd/main.go:368:7: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/statusd/main.go:430:5: missing defer call to LogOnPanic
[2024-10-22T17:17:34.655Z] /home/jenkins/workspace/status-go_prs_tests_PR-5969/cmd/statusd/main.go:509:5: missing defer call to LogOnPanic
```

igor-sirotin committed:
chore_: lint-fix

ilmotta commented on cmd/lint-panics/gopls/gopls.go:
```diff
@@ -0,0 +1,153 @@
+package gopls
```
@igor-sirotin, I'm under the impression the `gopls/` directory is mostly boilerplate and that the code is unrelated to the panic linter. Wouldn't it be better to move this whole package out of the `lint-panics/` directory? That way we facilitate the creation of new linters in the future by keeping the language server part decoupled a bit more.

igor-sirotin replied:
Yeah, it is quite some boilerplate, I was surprised that I had to write this. I'm still unsure that I did it right here, but didn't find any better way.

We can extract it, but same as for `parseFile`, I'd rather do that when it's needed, e.g. when we decided to write a new linter or smth.

ilmotta commented on cmd/lint-panics/analyzer/analyzer.go:
```diff
@@ -0,0 +1,247 @@
+	}
+}
+
+func (p *Analyzer) parseFile(path string, pass *analysis.Pass) (*ast.File, error) {
```
Could this be somewhere else to reduce friction to implement a new linter in the future? Seems reusable, not specific of the panic linter.

Well, I guess it might be better to wait for another custom linter to decide on what/how to best split and reuse code.

igor-sirotin replied:
Yeah, better to wait for the need for it 👍 
Also, this one is quite small, so I'm not sure if we need to extract it.

ilmotta reviewed (approved):
This is super cool @igor-sirotin 💫 I did a few quick checks, works pretty well!

I use gopls and it works flawlessly for me, but when I removed calls to defer `LogOnPanic()` I didn't see errors from gopls in my editor, only via `make lint` it works. Is the in-editor experience with gopls working for you?

qfrank commented on cmd/lint-panics/gopls/client.go:
maybe rename to `dummy_client.go` ?

qfrank commented on cmd/lint-panics/utils/utils.go:
```diff
@@ -0,0 +1,44 @@
+	"go.uber.org/zap/zapcore"
+)
+
+func PositionURI(pos gotoken.Position) string {
```
no usage?

igor-sirotin replied:
Indeed

qfrank commented on cmd/lint-panics/gopls/gopls.go:
```diff
@@ -0,0 +1,153 @@
+
+	client := NewDummyClient(logger)
+
+	// Create a JSON-RPC connection using stdin and stdout
```
// Step 1: ?

qfrank reviewed (approved):
Awesome job although there is an issue with the test. 🚀

igor-sirotin commented:
> ... when I removed calls to defer LogOnPanic() I didn't see errors from gopls in my editor, only via make lint it works. 
> Is the in-editor experience with gopls working for you?

@ilmotta This one is not integrated into IDE at all, that would require some more work. 
Good news is that I implemented it with `Analyzer` interface, so it can be integrated into `golangci-lint` or any other linter runners, which I guess includes IDEs. 

But again, this would require more work and I'm not sure if we need it right now. 
We can consider to do it if we encounter many-many such errors on PR level which would slow down development. But I doubt that this particular check will lead to such problems. Maybe some future linters will, but that's in the future 😄

igor-sirotin committed:
fix_: tests, address pr comments

osmaczko commented on services/wallet/api.go:
```diff
@@ -18,6 +18,7 @@ import (
 	signercore "github.com/ethereum/go-ethereum/signer/core/apitypes"
 	abi_spec "github.com/status-im/status-go/abi-spec"
 	"github.com/status-im/status-go/account"
+	common2 "github.com/status-im/status-go/common"
```
```suggestion
	status_common "github.com/status-im/status-go/common"
```

osmaczko commented on healthmanager/blockchain_health_manager.go:
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"sync"
 
+	gocommon "github.com/status-im/status-go/common"
```
```suggestion
	status_common "github.com/status-im/status-go/common"
```

osmaczko commented on cmd/lint-panics/analyzer/analyzer.go:
```diff
@@ -0,0 +1,247 @@
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+
+	gopls2 "github.com/status-im/status-go/cmd/lint-panics/gopls"
```
```suggestion
	gopls2 "github.com/status-im/status-go/cmd/lint-panics/gopls"
```

Naming with numbers makes my eyes bleed :D

igor-sirotin replied:
lol, but what's your proposal? 😄 
<img width="595" alt="image" src="https://github.com/user-attachments/assets/c92bba27-1740-4582-9339-e977684b35cc">

osmaczko commented on cmd/lint-panics/analyzer/analyzer.go:
```diff
@@ -0,0 +1,247 @@
+}
+
+func (p *Analyzer) Run(pass *analysis.Pass) (interface{}, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```
Would be great to use the same context that comes from `main()`.

osmaczko commented on cmd/lint-panics/analyzer/analyzer.go:
```diff
@@ -0,0 +1,247 @@
+		return nil, errors.New("analyzer is not type *inspector.Inspector")
+	}
+
+	// Check if the node is a GoStmt (which represents a 'go' statement)
```
Comment is a bit misleading, where is the "check"?

osmaczko commented on cmd/lint-panics/analyzer/analyzer.go:
```diff
@@ -0,0 +1,247 @@
+	// Traverse the AST to find goroutines
+	inspected.Preorder(nodeFilter, func(n ast.Node) {
+		p.ProcessNode(ctx, pass, n)
+	})
```
```suggestion
	// Traverse the AST to find goroutines (GoStmt represents a 'go' statement)
	inspected.Preorder([]ast.Node{(*ast.GoStmt)(nil)}, func(n ast.Node) {
		p.ProcessNode(ctx, pass, n)
	})
```

osmaczko commented on cmd/lint-panics/analyzer/analyzer.go:
```diff
@@ -0,0 +1,247 @@
+	return defPosition, p.checkGoroutine(body)
+}
+
+func (p *Analyzer) logLinterError(pass *analysis.Pass, errPos gotoken.Pos, callPos gotoken.Pos, err error) {
```
Is this piece of code responsile for the final return value from the lint-panics utility? I was struggling to find where and what lint-panics returns. If so, then I suggest to:

```suggestion
func (p *Analyzer) reportLinterError(pass *analysis.Pass, errPos gotoken.Pos, callPos gotoken.Pos, err error) {
```

igor-sirotin replied:
Yes, this is the place of reporting, I will rename.
Reporting is done with `pass.Report(...)`. This is the standard way in `go/analysis` package to report issues. 

We launch the analyzer here:
https://github.com/status-im/status-go/blob/f718e5d645f591f62bc4a25cef434664e2b1688c/cmd/lint-panics/main.go#L34

And in the end it automatically prints all reported issues.

osmaczko commented on Makefile:
```diff
@@ -408,7 +408,10 @@ canary-test: node-canary
 
-lint: generate
+lint-panics: generate
+	go run ./cmd/lint-panics -root="$(call sh, pwd)" -skip=./cmd -test=false ./...
```
I don't see a `test` flag defined in config :thinking:

igor-sirotin replied:
It's defined by the `go/analysis` package inside `singlechecker.Main` call:
https://github.com/status-im/status-go/blob/f718e5d645f591f62bc4a25cef434664e2b1688c/cmd/lint-panics/main.go#L34

If you run `go run ./cmd/lint-panics --help` you'll get a lot of fancy flags:
```
(venv) [23/10/24 8:24:08] ➜  status-go git:(feat/LogOnPanic-linter) ✗ go run ./cmd/lint-panics -help
logpanics: reports missing defer call to LogOnPanic

Usage: logpanics [-flag] [package]


Flags:
  -V    print version and exit
  -all
        no effect (deprecated)
  -c int
        display offending line with this many lines of context (default -1)
  -cpuprofile string
        write CPU profile to this file
  -debug string
        debug flags, any subset of "fpstv"
  -fix
        apply all suggested fixes
  -flags
        print analyzer flags in JSON
  -json
        emit JSON output
  -memprofile string
        write memory profile to this file
  -root string
        root directory to run gopls (default "/Users/igorsirotin/Repositories/Status/status-go")
  -skip string
        skip paths with this suffix
  -source
        no effect (deprecated)
  -tags string
        no effect (deprecated)
  -test
        indicates whether test files should be analyzed, too (default true)
  -trace string
        write trace log to this file
  -v    no effect (deprecated)
```

These are the standard flags for a go analyzer. With our specific flags added.

osmaczko commented on cmd/lint-panics/gopls/stream.go:
```diff
@@ -0,0 +1,29 @@
+import "io"
+
+// CombinedReadWriteCloser combines stdin and stdout into one interface.
+type CombinedReadWriteCloser struct {
```
```suggestion
type IOStream struct {
```

osmaczko reviewed (approved):
Very nice engineering!

I am concerned that a 5-minute context might be too short for fresh installs. It's probably faster for you because your gopls already has a cache and index built for status-go. Could you check how long it takes for a fresh repo?

igor-sirotin commented:
> I am concerned that a 5-minute context might be too short for fresh installs. It's probably faster for you because your gopls already has a cache and index built for status-go. Could you check how long it takes for a fresh repo?

@osmaczko I don't think it will be a problem. In fact, I think 5 minutes is way too much.
We run fresh `gopls` server, so I'd expect it to have 0 cache.  

But let me try 👌

igor-sirotin committed:
fix_: address pr comments
